site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -1623,15 +1624,17 @@ public function saveComponentsPage(Gradeable $gradeable, int $page) {
 
     /**
      * Route for adding a new component to a gradeable
-     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/components/new", methods={"POST"})
+     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/components/{peer}/new", methods={"POST"})
```

Comment:
This is improper usage of the route. Setting of the `$peer` variable should be done through a `$_POST` variable, not as a part of the route.